lib/commands/cloud-projects.ts

```diff
@@ -27,21 +25,28 @@ class ProjectIdCommandParameter implements ICommandParameter {
 
 export class CloudListProjectsCommand implements ICommand {
 	constructor(private $logger: ILogger,
-		private $remoteProjectService: IRemoteProjectService) { }
+		private $remoteProjectService: IRemoteProjectService,
+		private $prompter: IPrompter,
+		private $errors: IErrors) { }
 
-	allowedParameters: ICommandParameter[] = [];
+	allowedParameters: ICommandParameter[] = [new SolutionIdCommandParameter(this.$remoteProjectService)];
 
-	private printProjects(projects: any) {
-		this.$logger.out("Projects:");
-		projects.forEach((project: any, index: number) => {
-			this.$logger.out("%s: '%s'", (index + 1).toString(), project.name);
+	private printList(solutionName: string, list: Server.IWorkspaceItemData[]): void {
+		this.$logger.out(`Projects for ${solutionName} solution:`);
+		list.forEach((item: any, index: number) => {
+			this.$logger.out("%s: '%s'", (index + 1).toString(), item.Name);
 		});
 	}
 
 	execute(args: string[]): IFuture<void> {
 		return (() => {
-			let data = this.$remoteProjectService.getProjects().wait();
-			this.printProjects(data);
+			let projects: Server.IWorkspaceItemData[];
+			
+			let slnName = args[0] ? this.$remoteProjectService.getSolutionName(args[0]).wait() : 
```

Comment:
This code is quite unreadable. Consider writing it as an "if". And I am a guy who adores the ternary operator :)